nemo_skills/inference/reward_model.py

```diff
@@ -52,20 +47,24 @@ class RewardModelConfig:
     # Used to identify the input file if `input_dir` is provided. If `random_seed` is not provided,
     # the input will be assumed to be from 'greedy' generation
     random_seed: str | None = None
-    batch_size: int = 128
-    max_samples: int = -1  # If > 0, will stop after generating this many samples. Useful for debugging
-    skip_filled: bool = False  # If True, will skip the generations that are already in the output file
-
-    # if > 0, will skip this many samples from the beginning of the data file.
-    # Useful if need to run multiple slurm jobs on the same data file
-    offset: int = 0
-    # Default reward model type
-    reward_model_type: str = "orm"
-    reward_model_score_key: str = "reward_model_score"
+    # Inheritance was converting these dataclasses to dicts, so to be on the safe side we override them
+    inference: InferenceConfig = field(default_factory=InferenceConfig)  # LLM call parameters
+    # Inference server configuration {server_params}
+    server: dict = field(default_factory=dict)
+    # Sandbox configuration {sandbox_params}
+    sandbox: dict = field(default_factory=dict)
 
-    # can add this flag to just print the first prompt instead of running generation
-    # useful to double check that your data can be loaded and prompt has what you expect
-    dry_run: bool = False
+    # Async loop is currently not supported for reward model
```

Comment:
Similarly, we should ideally disallow usage of all code execution related parameters as they are unused in this class. We can probably add something like "disallow_parameters" or a similar method to the main class that will define a list of parameters (and perhaps new defaults?) and if any of those has value not equal to the new default (meaning it's provided by the user), we should throw an error. Otherwise there is a chance for silent misbehavior (user specifies some parameters and expect them to affect generations, but they are not used), which we should try to avoid as much as possible. But we can do it in another PR I guess, as it's unlikely anyone is going to use code execution with reward models and expect that to do something

Reply:
I'm adding `check_contamination.py` as well.

Reply:
@Kipok - Added the inheritance to `check_contaminatation.py` and a method to check the correctness of arguments.